itinerant-iterators/app/cli.py

```diff
@@ -0,0 +1,100 @@
+from pathlib import Path
+
+import click
+from PIL import Image
+from rich.console import Console
+
+from app import obfuscate, steganography
+
+from .ui import main as ui_main
+
+
+@click.group(invoke_without_command=False)
+@click.pass_context
+def main(ctx):
+    """Main entry point to application"""
+    pass
+
+
+@main.command()
+def ui():
+    """Main entry point to the ui"""
+    ui_main()
+
+
+@main.group(invoke_without_command=False)
+def cli():
+    """Main entry point to the cli"""
+    pass
+
+
+@cli.group(invoke_without_command=False)
+def watermark():
+    """Entrypoint to watermark subcommand"""
+    pass
+
+
+@watermark.command("encode")
+@click.argument("text", type=str)
+@click.argument(
+    "infile",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+)
+@click.argument(
+    "outfile",
+    type=click.Path(exists=False, dir_okay=False, file_okay=True, writable=True),
+)
+def watermark_encode(text: str, infile: Path, outfile: Path):
+    """Entrypoint to watermark encode command"""
+    steganography.encode(text, infile, outfile, steganography.Lsb())
```

Comment:
This is purely stylistic, but using keyword arguments is often more readable for booleans and other "magic values". 

```suggestion
    steganography.encode(text, infile, outfile, algorithm=steganography.Lsb())
```